libmamba/include/mamba/core/fetch.hpp

```diff
@@ -61,9 +138,9 @@ namespace mamba
         curl_off_t get_speed();
 
         template <class C>
-        inline void set_finalize_callback(bool (C::*cb)(), C* data)
+        inline void set_finalize_callback(bool (C::*cb)(const DownloadTarget&), C* data)
```

Comment:
Why not just take anything that can be stored in that `std::function` (which already checks for the signature) and not bother about what type is `C`, let the caller setup the right thing (lambda capturing etc.)?

Reply:
That's what we do in the powerloader, actually. Initially I didn't want to change the interfaces too much ...

Reply:
Ah yes that felt familiar XD
Ok then it will be changed later.